gamemodes/terrortown/gamemode/client/cl_targetid.lua

```diff
@@ -837,7 +837,17 @@ function HUDDrawTargetIDRagdolls(tData)
 	)
 
 	if tData:GetEntityDistance() <= 100 then
-		tData:SetSubtitle(ParT("corpse_hint", key_params))
+		if GetConVar("ttt2_inspect_detective_only"):GetBool() and client:GetBaseRole() ~= ROLE_DETECTIVE then
+			if client:IsActive() and client:IsShopper() and CORPSE.GetCredits(ent, 0) > 0 then
```

Comment:
`client:IsActive()` intended? Or do you mean `client:Alive() and client:IsTerror()`?

Reply:
It has been this way eversince. Changing to your suggestion might actually break some things.